.github/workflows/tests.yml

```diff
@@ -87,15 +87,19 @@ jobs:
     - name: Install dependencies
       run: |
         set -x
-        pip install --upgrade pip tox codecov
+        python3 -m venv venv
+        . ./venv/bin/activate
```

Comment:
I suppose the reasoning behind venv is in this fix, could you please elaborate on what the problem was?